### PR TITLE
Add golint, make it pass cleanly, fix some docs

### LINKED ISSUE #669: Golint and Godoc issues
Both golint and godoc.org show some issues in packages' documentation. The linter mostly complains about "malformed" or missing doc comments, e.g. `notify.go:63:1: comment on exported function NewListenerConn should be of the form "NewListenerConn ..."`. It also complains about package `listen_example`'s name containing an underscore. Godoc is mostly fine, aside from the missing comments, but some code and lists are not aligned to the 80 char mark, and thus the scroll bars appears at https://godoc.org/github.com/lib/pq#pkg-constants. Thus, some proposals:

* Align all code and lists to the 80 char mark.

* Add missing documentation and fix documentation golint complains about.

* Move `listen_example` to `examples/listen`.

* Add a link to godoc.org to the README.

Adding golint to the Travis builds is doable, but I am not sure if it's worth it.

Thoughts?

### PR DESCRIPTION
**Pull request by ainar-g**
This change introduces golint to the TravisCI build process, fixes everything it has found, and additionally fixes some Godoc scrolling isuues.

Fixes #669.

### CODE REVIEW

File: conn.go [hunk 1]
```diff
@@ -1432,7 +1441,8 @@ func (rs *rows) NextResultSet() error {
 //
 //    tblname := "my_table"
 //    data := "my_data"
-//    err = db.Exec(fmt.Sprintf("INSERT INTO %s VALUES ($1)", pq.QuoteIdentifier(tblname)), data)
+//    quoted := pq.QuoteIdentifier(tblname)
+//    err = db.Exec(fmt.Sprintf("INSERT INTO %s VALUES ($1)", quoted), data)
```
Comment (tamird) [inline conn.go]: shouldn't `err` be `:=`?

File: conn.go [hunk 2]
```diff
@@ -78,6 +82,7 @@ func (s transactionStatus) String() string {
 	panic("not reached")
 }
 
+// Dialer is the dialer interface.
```
Comment (tamird) [inline conn.go]: This comment isn't very helpful.

File: notify.go [hunk 3]
```diff
@@ -346,40 +347,48 @@ func (l *ListenerConn) Close() error {
...
 var errListenerClosed = errors.New("pq: Listener has been closed")
 
+// ErrChannelAlreadyOpen is returned from Listen when a channel is already
+// open.
 var ErrChannelAlreadyOpen = errors.New("pq: channel is already open")
+
+// ErrChannelNotOpen is returned from Unlisten when a channel is not open.
 var ErrChannelNotOpen = errors.New("pq: channel is not open")
 
+// ListenerEventType describes the type of the listener event.
```
Comment (tamird) [inline notify.go]: > ListenerEventType is an enumeration of listener event types.

File: notify.go [hunk 4]
```diff
@@ -346,40 +347,48 @@ func (l *ListenerConn) Close() error {
...
 type ListenerEventType int
 
 const (
-	// Emitted only when the database connection has been initially
-	// initialized.  err will always be nil.
+	// ListenerEventConnected is emitted only when the database connection
+	// has been initially initialized. err will always be nil.
```
Comment (tamird) [inline notify.go]: what err?

File: notify.go [hunk 5]
```diff
@@ -346,40 +347,48 @@ func (l *ListenerConn) Close() error {
...
-	// Emitted after a connection to the database was attempted, but failed.
-	// err will be set to an error describing why the connection attempt did
-	// not succeed.
+	// ListenerEventConnectionAttemptFailed is emitted after a connection
+	// to the database was attempted, but failed. err will be set to an
+	// error describing why the connection attempt did not succeed.
 	ListenerEventConnectionAttemptFailed
 )
 
+// EventCallbackType describes the event callback type.
```
Comment (tamird) [inline notify.go]: s/describes/is/

**Comment by ainar-g** [comment]
@tamird PTAL.

**Comment by tamird** [comment]
Thanks!

**Pull request merged**